### PR TITLE
[UI smoke test] External links pwa UI tests

### PR DESCRIPTION
**Pull request by AndiAJ**
New external links UI smoke tests:
• `externalLinkPWATest`
• `emailLinkPWATest`
• `telephoneLinkPWATest`

✅  Successfully passed 60x on Firebase

<details>

<summary> 📈 Results (dropdown)</summary>

matrix | result | logs | details
-- | -- | -- | --
matrix-2nrwz3kjeyalj | success | logs | 3 test cases passed
matrix-16hagkcyr6cbb | success | logs | 3 test cases passed
matrix-kzewh87xx4tka | success | logs | 3 test cases passed
matrix-qaboeiaiu277a | success | logs | 3 test cases passed
matrix-2wtkfplxidtj3 | success | logs | 3 test cases passed
matrix-290ecwvgcdd13 | success | logs | 3 test cases passed
matrix-1e2pvsi8n6ka1 | success | logs | 3 test cases passed
matrix-2a72glsot9f50 | success | logs | 3 test cases passed
matrix-xjxs7baqtw5pa | success | logs | 3 test cases passed
matrix-3fjtksc0lpcld | success | logs | 3 test cases passed
matrix-2a2o5pfiav1gj | success | logs | 3 test cases passed
matrix-1819bgs2ux38k | success | logs | 3 test cases passed
matrix-3au6me1uu0qml | success | logs | 3 test cases passed
matrix-2pkxqkz657xw0 | success | logs | 3 test cases passed
matrix-2uxzl0ip73gls | success | logs | 3 test cases passed
matrix-3lhjneq774amo | success | logs | 3 test cases passed
matrix-19ti8z71bi2at | success | logs | 3 test cases passed
matrix-ix89oujofpooa | success | logs | 3 test cases passed
matrix-wqbrhlipsbtxa | success | logs | 3 test cases passed
matrix-2mnz9w2mgc7ay | success | logs | 3 test cases passed
matrix-38c0ur7e1emcn | success | logs | 3 test cases passed
matrix-37owuy7ixlwey | success | logs | 3 test cases passed
matrix-235evzuwqvl7b | success | logs | 3 test cases passed
matrix-3tdkmrwj7tosv | success | logs | 3 test cases passed
matrix-1at0t6i6cf18a | success | logs | 3 test cases passed
matrix-t29imfjr5q4qa | success | logs | 3 test cases passed
matrix-1pwqbs8vrjs6m | success | logs | 3 test cases passed
matrix-27a2vxrnsuwwp | success | logs | 3 test cases passed
matrix-u3u2e23jy38na | success | logs | 3 test cases passed
matrix-1y9rv6rx6oi76 | success | logs | 3 test cases passed
matrix-1be3xr2qjirrj | success | logs | 3 test cases passed
matrix-fn2qz870g4lta | success | logs | 3 test cases passed
matrix-3kqj7jdh88b5i | success | logs | 3 test cases passed
matrix-3vu48ke9vta0y | success | logs | 3 test cases passed
matrix-23hmsmoq5pgw1 | success | logs | 3 test cases passed
matrix-3r95dwyids3tq | success | logs | 3 test cases passed
matrix-1vbcxl82aznyn | success | logs | 3 test cases passed
matrix-3tlxd2z91vfy9 | success | logs | 3 test cases passed
matrix-hhcccgec3om8a | success | logs | 3 test cases passed
matrix-1qynhkcqwd0jc | success | logs | 3 test cases passed
matrix-1lnyq9eadmbv1 | success | logs | 3 test cases passed
matrix-116ekcgx364g3 | success | logs | 3 test cases passed
matrix-39zdlewbfdtmp | success | logs | 3 test cases passed
matrix-1c2hft2s6msk0 | success | logs | 3 test cases passed
matrix-2iq8xsrug0tnp | success | logs | 3 test cases passed
matrix-qr9a9ez5zjxra | success | logs | 3 test cases passed
matrix-1lhpyb2zrqko5 | success | logs | 3 test cases passed
matrix-28rfclqwx3nmv | success | logs | 3 test cases passed
matrix-2sd7g3ln0bap7 | success | logs | 3 test cases passed
matrix-7933yv5tfyqsa | success | logs | 3 test cases passed
matrix-ixroskf21nsaa | success | logs | 3 test cases passed
matrix-2yc6d9z2xo9zx | success | logs | 3 test cases passed
matrix-3ou4jvhs4yqvt | success | logs | 3 test cases passed
matrix-caxmuylxtppaa | success | logs | 3 test cases passed
matrix-m7m8aowwl62ha | success | logs | 3 test cases passed

</details>

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/androidTest/java/org/mozilla/fenix/ui/PwaTest.kt [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import org.mozilla.fenix.helpers.FeatureSettingsHelper
+import org.mozilla.fenix.helpers.HomeActivityIntentTestRule
+import org.mozilla.fenix.helpers.TestHelper.assertNativeAppOpens
+import org.mozilla.fenix.ui.robots.customTabScreen
+import org.mozilla.fenix.ui.robots.navigationToolbar
+
+class PwaTest {
+    private val featureSettingsHelper = FeatureSettingsHelper()
+    private val externalLinksPWAPage = "https://andiaj.github.io/testapp/externalLinks.html"
+    private val emailLink = "mailto://notta-r3al-3mail@gmail.com"
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/PwaTest.kt:20]: Lets use `example@example.com`

File: app/src/androidTest/java/org/mozilla/fenix/ui/PwaTest.kt [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    private val phoneLink = "tel://+1-234-555-1212"
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/PwaTest.kt:21]: Let's use `tel://123456789`

**Review by AaronMT — CHANGES_REQUESTED** [review]

File: app/src/androidTest/java/org/mozilla/fenix/ui/PwaTest.kt [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+import org.mozilla.fenix.helpers.Constants.PackageName.PHONE_APP
+import org.mozilla.fenix.helpers.FeatureSettingsHelper
+import org.mozilla.fenix.helpers.HomeActivityIntentTestRule
+import org.mozilla.fenix.helpers.TestHelper.assertNativeAppOpens
+import org.mozilla.fenix.ui.robots.customTabScreen
+import org.mozilla.fenix.ui.robots.navigationToolbar
+
+class PwaTest {
+    private val featureSettingsHelper = FeatureSettingsHelper()
+    private val externalLinksPWAPage = "https://andiaj.github.io/testapp/externalLinks.html"
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/PwaTest.kt:19]: Reminder to change this

**Review by AaronMT — CHANGES_REQUESTED** [review]

File: app/src/androidTest/java/org/mozilla/fenix/ui/PwaTest.kt (hunk 1 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/PwaTest.kt]: Done !

File: app/src/androidTest/java/org/mozilla/fenix/ui/PwaTest.kt (hunk 2 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/PwaTest.kt]: Done !

**Comment by AndiAJ** [comment]
Hi @AaronMT , made the changes and each test successfully passed 20x on Firebase.

The last thing that I'll need to do before landing will be to change the `externalLinksPWAPage` link

**Review by AaronMT — APPROVED** [review]

File: app/src/androidTest/java/org/mozilla/fenix/ui/PwaTest.kt (hunk 3 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/PwaTest.kt]: Done

**Pull request merged**